modules/calcite/src/main/java/org/apache/ignite/internal/processors/query/calcite/rule/logical/ExposeIndexRule.java

```diff
@@ -71,7 +71,10 @@ private static boolean preMatch(IgniteLogicalTableScan scan) {
             .map(idx -> idx.toRel(cluster, optTable, proj, condition, requiredCols))
             .collect(Collectors.toList());
 
-        assert !indexes.isEmpty();
+        indexes.removeIf(Objects::isNull);
```

Comment:
perhaps, it would be a bit better to filter out null with a stream API `.filter(Objects::nonNull)` before collecting them to a `indexes` collection

Reply:
This block was rewritten, so there is no null check is required now.